modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ElmClientCodegen.java

```diff
@@ -474,6 +471,19 @@ private ElmImport createImport(final String name) {
                     }
                 }
             }
+
+            hasDate = hasDate
+                || op.queryParams.stream().filter(param -> param.isDate).findAny().isPresent()
+                || op.pathParams.stream().filter(param -> param.isDate).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isDate).findAny().isPresent();
+            hasDateTime = hasDate
+                || op.queryParams.stream().filter(param -> param.isDateTime).findAny().isPresent()
+                || op.pathParams.stream().filter(param -> param.isDateTime).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isDateTime).findAny().isPresent();
+            hasUuid = hasUuid
+                || op.queryParams.stream().filter(param -> param.isUuid).findAny().isPresent()
+                || op.pathParams.stream().filter(param -> param.isUuid).findAny().isPresent()
+                || op.headerParams.stream().filter(param -> param.isUuid).findAny().isPresent();
```

Comment:
Not sure about the types in java, but you can probably combine all params (query, path, header) into a single collection or stream first, and filter afterwards only once, can you? ;)

Reply:
You're absolutely right this isn't very clean. I've updated the PR.

Reply:
👍